src/NuGet.Core/NuGet.Protocol/Plugins/RequestHandlers/GetCredentialsRequestHandler.cs

```diff
@@ -154,19 +155,29 @@ public void AddOrUpdateSourceRepository(SourceRepository sourceRepository)
                         cancellationToken);
                 }
 
-                if (credential == null)
+                if (credential is AuthTypeFilteredCredentials filteredCredentials)
                 {
                     responsePayload = new GetCredentialsResponse(
-                        MessageResponseCode.NotFound,
-                        username: null,
-                        password: null);
+                        MessageResponseCode.Success,
+                        filteredCredentials.InnerCredential.UserName,
+                        filteredCredentials.InnerCredential.Password,
+                        filteredCredentials.AuthTypes);
                 }
-                else
+                else if (credential is NetworkCredential networkCredential)
                 {
                     responsePayload = new GetCredentialsResponse(
                         MessageResponseCode.Success,
-                        credential.UserName,
-                        credential.Password);
+                        networkCredential.UserName,
+                        networkCredential.Password);
+                }
+                else
+                {
+                    networkCredential = credential?.GetCredential(packageSource.SourceUri, null);
```

Comment:
Just curious what are the cases where `credential` is neither `NetworkCredential` nor `AuthTypeFilteredCredential`?

Reply:
This comment prompted me to take another look and catch this issue: 

https://github.com/NuGet/NuGet.Client/pull/2297/files/12e5bf27e166727b9c9ce80b51d76f86d57d7611#r207999444